### DISCUSSION TITLE
could the project support the strategy ?

### DISCUSSION THREAD

**Discussion by ChainCoderX** [rodrigo-brito/ninjabot#99] [Q&A] [answered]
////////////////////////////////////////////////////////////
//  Copyright by HPotter v1.0 20/06/2014
// The Moving Average Crossover trading strategy is possibly the most popular
// trading strategy in the world of trading. First of them were written in the
// middle of XX century, when commodities trading strategies became popular.
// This strategy is a good example of so-called traditional strategies. 
// Traditional strategies are always long or short. That means they are never 
// out of the market. The concept of having a strategy that is always long or 
// short may be scary, particularly in today’s market where you don’t know what 
// is going to happen as far as risk on any one market. But a lot of traders 
// believe that the concept is still valid, especially for those of traders who 
// do their own research or their own discretionary trading. 
// This version uses crossover of moving average and its exponential moving average. 
////////////////////////////////////////////////////////////
study(title="EMA & MA Crossover", shorttitle="EMA & MA Crossover", overlay = true)
LengthMA = input(10, minval=1)
LengthEMA = input(10,minval=1)
xMA = sma(close, LengthMA)
xEMA = ema(xMA, LengthEMA)
pos = iff(xEMA < xMA , 1,
	    iff(xEMA > xMA, -1, nz(pos[1], 0))) 
barcolor(pos == -1 ? red: pos == 1 ? green : blue)
plot(xMA, color=red, title="MA")
plot(xEMA, color=blue, title="EMA")

**Comment by ChainCoderX** [comment]
and how to plot the chart  on web page ?

**Comment by rodrigo-brito** [comment, marked as answer]
Hi @wade-liwei, Ninjabot supports custom strategies, you can create it with code.
We have an example of Crossover Strategy here: https://github.com/rodrigo-brito/ninjabot/blob/main/examples/strategies/emacross.go

You can adapt the code with your parameters.

**Reply by sohaha** [reply]
May I ask, how can I run the strategy in real-time?

**Reply by rodrigo-brito** [reply]
In the `examples` directory, we have some examples. Here you can execute in the Spot Market: https://github.com/rodrigo-brito/ninjabot/blob/main/examples/spotmarket/spot.go

**Reply by sohaha** [reply]
Thank you, I will take a look and execute the futures.
